recipes/das_node.rb

```diff
@@ -0,0 +1,485 @@
+    password_file password_file
+    username username
+    admin_port admin_port
+    target payara_config
+    asadmin asadmin
+    internal_port 28182
+    network_name "http-internal"
+    network_listener_name "http-int-list"
+    securityenabled false
+    action :glassfish_configure_network
+  end
+end
+
+glassfish_asadmin "set #{payara_config}.http-service.virtual-server.server.property.send-error_1='code=404 path=${com.sun.aas.instanceRoot}/docroot/index.html reason=Resource_not_found'" do
+  domain_name domain_name
+  password_file password_file
+  username username
+  admin_port admin_port
+  secure false
+end
+
+# disable monitoring and http-listeners on server-config
+glassfish_network_listener_conf = {
+  "configs.config.server-config.network-config.network-listeners.network-listener.http-listener-2.enabled" => false,
+  "configs.config.server-config.network-config.network-listeners.network-listener.https-int-list.enabled" => false,
+  "configs.config.server-config.rest-monitoring-configuration.enabled" => false,
+  "configs.config.server-config.monitoring-service.mbean-enabled" => false,
+  "configs.config.server-config.monitoring-service.monitoring-enabled" => false,
+  "configs.config.server-config.microprofile-metrics-configuration.enabled" => false
+}
+
+hopsworks_configure_server "glassfish_configure" do
+  domain_name domain_name
+  domains_dir domains_dir
+  password_file password_file
+  username username
+  admin_port admin_port
+  target payara_config
+  asadmin asadmin
+  override_props glassfish_network_listener_conf
+  action :glassfish_configure
+end
+
+hopsworks_configure_server "glassfish_configure_monitoring" do
+  domain_name domain_name
+  domains_dir domains_dir
+  password_file password_file
+  username username
+  admin_port admin_port
+  target payara_config
+  asadmin asadmin
+  action :glassfish_configure_monitoring
+end
+
+node_ips=[public_ip]+config_nodes
+interfaces=node_ips.join(",")
+glassfish_asadmin "set-hazelcast-configuration --enabled true --publicaddress #{public_ip} --daspublicaddress #{public_ip} --autoincrementport true --interfaces #{interfaces} --membergroup #{payara_config} --target #{payara_config}" do
+  domain_name domain_name
+  password_file password_file
+  username username
+  admin_port admin_port
+  secure false
+end
+
+glassfish_asadmin "create-local-instance --config #{payara_config} --nodedir #{nodedir} #{local_instance}" do
+  domain_name domain_name
+  password_file password_file
+  username username
+  admin_port admin_port
+  secure false
+  not_if "#{asadmin_cmd} list-instances | grep #{local_instance}"
+end
+
+directory node['hopsworks']['data_volume']['localhost-domain1'] do
+  owner node['glassfish']['user']
+  group node['glassfish']['group']
+  mode '0750'
+end
+
+directory node['hopsworks']['data_volume']['node_logs'] do
+  owner node['glassfish']['user']
+  group node['glassfish']['group']
+  mode '0750'
+end
+
+bash 'Move glassfish logs to data volume' do
```

Comment:
This is not needed I think. The purpose of this was to move old directories to the data volume. If you already create the directory in the data volume and symlink it then this snippet is not needed.